protos/commands.proto

```diff
@@ -17,49 +17,102 @@ enum Type {
 	ARTIST_COUNT = 8;
 	FOLDER_COUNT = 9;
 	SELECT_QUERY = 10;
-	CONTENT_QUERY = 11;
 	WHEN_DEVICE = 12;
```

Comment:
Yeah I apologize, it might not be a thing for enums. Definitely use it for message values.
In case you haven't encountered it yet, the `oneof` keyword might be helpful. It allows for polymorphism and hacky generic-ness.